src/main/java/org/jenkinsci/plugins/workflow/steps/PwdStep.java

```diff
@@ -95,7 +95,11 @@ public boolean isTmp() {
 
         @Override protected String run() throws Exception {
             FilePath cwd = getContext().get(FilePath.class);
-            return (tmp ? WorkspaceList.tempDir(cwd) : cwd).getRemote();
+            if (tmp && cwd != null)
+                cwd = WorkspaceList.tempDir(cwd);
+            if (cwd == null)
```

Comment:
This should never happen because the Pipeline engine uses `DescriptorImpl.getRequiredContext` (which returns a list containing `FilePath.class`) to check that all required context variables are available before the step is executed, and if they are not available, it throws an exception. I would either suppress the warning, or add `assert cwd != null` or `Objects.requireNonNull(cwd)` right after the `getContext()` call and leave the old line the same as it was.

Reply:
`WorkspaceList.tempDir()` is marked as `@CheckForNull`, so even with the `cwd` param checked for null, that may still trip the spotbugs check. Will give it a try.

Reply:
Ah I forgot that `WorkspaceList.tempDir` is `@CheckForNull`. It should only return null if `cwd` does not have a parent directory. I think it would make sense to throw an exception if `tmp` is `true` and `cwd` has no parent directory since there is no temporary directory in that case (because it is supposed to be a sibling of the current directory).

Reply:
Added the `requireNonNull()`, and an `IOException` when `tempDir()` returns `null`. I did not add a Messages resources for that.